From 2b09029e48b3a855caee67a72706bb0a06a7548c Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Tue, 21 May 2024 09:58:39 +0530
Subject: [PATCH 1132/1141] phy: stm32: Remove support for internal-vbus-comp
 property

Remove the support for usbphy internal vbus comparator
as Vbus pin has been removed in RevB

Change-Id: Id801f1d7ef59dc4deeb5d6c5790d4fb64e3f3492
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/381176
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/phy/st/phy-stm32-usb2phy.c | 44 ++++++++++--------------------
 1 file changed, 14 insertions(+), 30 deletions(-)

diff --git a/drivers/phy/st/phy-stm32-usb2phy.c b/drivers/phy/st/phy-stm32-usb2phy.c
index 836bad72dfe2..375253531527 100644
--- a/drivers/phy/st/phy-stm32-usb2phy.c
+++ b/drivers/phy/st/phy-stm32-usb2phy.c
@@ -81,7 +81,7 @@ struct stm32_usb2phy {
 	struct regulator *vdd33, *vdda18;
 	enum phy_mode mode;
 	u32 mask_trim1, value_trim1, mask_trim2, value_trim2;
-	bool internal_vbus_comp, is_init;
+	bool is_init;
 	struct clk_hw clk48_hw;
 	atomic_t en_refcnt;
 	const struct stm32mp2_usb2phy_hw_data *hw_data;
@@ -351,7 +351,7 @@ static int stm32_usb2phy_set_mode(struct phy *phy, enum phy_mode mode, int submo
 						 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK,
 						 0);
 		else {
-			if (!phy_dev->internal_vbus_comp && submode == USB_ROLE_NONE) {
+			if (submode == USB_ROLE_NONE) {
 				ret = regmap_update_bits(phy_dev->regmap,
 							 phy_data->cr_offset,
 							 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
@@ -374,35 +374,25 @@ static int stm32_usb2phy_set_mode(struct phy *phy, enum phy_mode mode, int submo
 		break;
 
 	case PHY_MODE_USB_DEVICE:
-		if (phy_dev->internal_vbus_comp) {
+		if (submode == USB_ROLE_NONE) {
 			ret = regmap_update_bits(phy_dev->regmap,
 						 phy_data->cr_offset,
 						 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
 						 SYSCFG_USB2PHY2CR_VBUSVALID_MASK |
 						 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK |
 						 SYSCFG_USB2PHY2CR_VBUSVLDEXT_MASK,
-						 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK);
+						 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
+						 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK);
 		} else {
-			if (submode == USB_ROLE_NONE) {
-				ret = regmap_update_bits(phy_dev->regmap,
-							 phy_data->cr_offset,
-							 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVALID_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVLDEXT_MASK,
-							 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK);
-			} else {
-				ret = regmap_update_bits(phy_dev->regmap,
-							 phy_data->cr_offset,
-							 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVALID_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVLDEXT_MASK,
-							 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK |
-							 SYSCFG_USB2PHY2CR_VBUSVLDEXT_MASK);
-			}
+			ret = regmap_update_bits(phy_dev->regmap,
+						 phy_data->cr_offset,
+						 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
+						 SYSCFG_USB2PHY2CR_VBUSVALID_MASK |
+						 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK |
+						 SYSCFG_USB2PHY2CR_VBUSVLDEXT_MASK,
+						 SYSCFG_USB2PHY2CR_USB2PHY2CMN_MASK |
+						 SYSCFG_USB2PHY2CR_VBUSVLDEXTSEL_MASK |
+						 SYSCFG_USB2PHY2CR_VBUSVLDEXT_MASK);
 		}
 		if (ret) {
 			dev_err(dev, "can't set usb2phycr (%d)\n", ret);
@@ -751,12 +741,6 @@ static int stm32_usb2phy_probe(struct platform_device *pdev)
 	phy_dev->phy = phy;
 	phy_set_drvdata(phy, phy_dev);
 
-	if (phy_dev->hw_data->valid_mode != USB2_MODE_HOST_ONLY) {
-		phy_dev->internal_vbus_comp = of_property_read_bool(np, "st,internal-vbus-comp");
-		dev_dbg(dev, "Using USB2PHY %s VBUS Comparator\n",
-			phy_dev->internal_vbus_comp ? "Internal" : "External");
-	}
-
 	/* Configure phy tuning */
 	ret = stm32_usb2phy_tuning(phy);
 	if (ret)
-- 
2.39.2

